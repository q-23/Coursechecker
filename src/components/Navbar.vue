<template>
  <div>
    <v-app-bar
        color="deep-purple accent-4"
        class="mb-16"
        dense
        dark
    >
      <v-toolbar-title>NBP Course Check</v-toolbar-title>
      <v-spacer/>
      <router-link to="/">
        <div class="my-2 mx-2">
          <v-btn small color="primary">All</v-btn>
        </div>
      </router-link>
      <router-link to="/favourite">
        <div class="my-2 mx-2">
          <v-btn small color="pink">Favs</v-btn>
        </div>
      </router-link>
    </v-app-bar>
  </div>
</template>