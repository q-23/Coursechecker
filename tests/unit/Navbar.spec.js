import Vue from "vue";
import Vuetify from 'vuetify'
import VueRouter from "vue-router";

import { shallowMount, createLocalVue } from '@vue/test-utils'

import Navbar from "@/components/Navbar";

const localVue = createLocalVue()
const router = new VueRouter()

Vue.use(Vuetify)
localVue.use(VueRouter)

describe('[NAVBAR]', () => {
	let vuetify;

	beforeEach(() => {
		vuetify = new Vuetify()
	})

	it('Should render component', () => {
		const wrapper = shallowMount(Navbar, {
			localVue,
			vuetify,
			router
		})
		expect(wrapper).toMatchSnapshot()
	})

	it('Should render app name', () => {
		const name = 'NBP Course Check'
		const wrapper = shallowMount(Navbar, {
			localVue,
			vuetify,
			router
		})
		expect(wrapper.text()).toMatch(name)
	})
})
