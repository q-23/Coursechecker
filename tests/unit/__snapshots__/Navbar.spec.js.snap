// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[NAVBAR] Should render component 1`] = `
<div>
  <v-app-bar-stub color="deep-purple accent-4" dark="true" tag="header" dense="true" extensionheight="48" src="" value="true" class="mb-16">
    <v-toolbar-title-stub>NBP Course Check</v-toolbar-title-stub>
    <v-spacer-stub></v-spacer-stub>
    <router-link-stub to="/" tag="a" ariacurrentvalue="page" event="click">
      <div class="my-2 mx-2">
        <v-btn-stub color="primary" tag="button" activeclass="" small="true" type="button">All</v-btn-stub>
      </div>
    </router-link-stub>
    <router-link-stub to="/favourite" tag="a" ariacurrentvalue="page" event="click">
      <div class="my-2 mx-2">
        <v-btn-stub color="pink" tag="button" activeclass="" small="true" type="button">Favs</v-btn-stub>
      </div>
    </router-link-stub>
  </v-app-bar-stub>
</div>
`;
